### PR TITLE
API Platform 3

### PR DESCRIPTION
**Pull request by rimi-itk**
#### Link to ticket

https://jira.itkdev.dk/browse/DISPLAY-1030

#### Description

Upgrades to API Platform 3.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [x] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

## Notes on API spec

* The route `authentication-token` (https://github.com/os2display/display-api-service/blob/develop/config/routes.yaml#L8-L11) suddenly appears in the API spec (!)

### CODE REVIEW

**Comment by rimi-itk** [comment]
@tuj, please review this.

File: src/State/AbstractProcessor.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    public function toOutput(object $object): object
+    {
+        return null !== $this->provider ? $this->provider->toOutput($object) : $object;
+    }
+
+    /**
+     * Load previous object if any.
+     *
+     * This is needed to get an objet handled by entity manager.
```
Comment (tuj) [inline src/State/AbstractProcessor.php:52]: "object"

File: src/Entity/AbstractBaseEntity.php [hunk 2]
```diff
@@ -76,7 +75,8 @@ public function getCreatedAt(): \DateTimeInterface
     /**
      * @ORM\PrePersist()
      */
-    public function setCreatedAtValue(): self
+    // FIXME Why can't I #[Ignare] this function?
```
[the remark this answers is not in the document]
Comment (tuj) [reply src/Entity/AbstractBaseEntity.php:78]: "Ignore"

File: src/Entity/AbstractBaseEntity.php [hunk 3]
```diff
@@ -93,7 +93,8 @@ public function getModifiedAt(): \DateTimeInterface
      *
      * @ORM\PreUpdate()
      */
-    public function setModifiedAtValue(): self
+    // FIXME Why can't I #[Ignare] this function?
```
Comment (tuj) [inline src/Entity/AbstractBaseEntity.php:96]: "Ignore"

File: tests/Api/SlidesTest.php [hunk 4]
```diff
@@ -46,21 +46,23 @@ public function testGetItem(): void
...
-                'modified' => 'Slide/modified',
-                'modifiedBy' => 'Slide/modifiedBy',
-                'createdBy' => 'Slide/createdBy',
-                'templateInfo' => 'Slide/templateInfo',
-                'onPlaylists' => 'Slide/onPlaylists',
-                'duration' => 'Slide/duration',
-                'published' => 'Slide/published',
-                'content' => 'Slide/content',
-            ],
+            // FIXME: What is the correct context?
```
Comment (tuj) [inline tests/Api/SlidesTest.php:49]: Remove this commented out code. Let's ignore context in tests

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/Entity/AbstractBaseEntity.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (rimi-itk) [reply src/Entity/AbstractBaseEntity.php]: I need (new) glasses! https://github.com/os2display/display-api-service/pull/173/commits/985bd3cade1c8c326dcd630ee3f99cfc79664417

File: src/Entity/AbstractBaseEntity.php (hunk 3 above)
Comment (rimi-itk) [reply src/Entity/AbstractBaseEntity.php]: I need (new) glasses! https://github.com/os2display/display-api-service/pull/173/commits/985bd3cade1c8c326dcd630ee3f99cfc79664417

File: src/State/AbstractProcessor.php (hunk 1 above)
Comment (rimi-itk) [reply src/State/AbstractProcessor.php]: Fixed in https://github.com/os2display/display-api-service/pull/173/commits/504d67ed15c0ce1a341af7d8147666e5ff6fcc74.

File: src/Dto/ScreenCampaign.php [hunk 5]
```diff
@@ -2,11 +2,18 @@
...
 
+use App\Dto\Trait\BlameableTrait;
+use App\Dto\Trait\IdentifiableTrait;
+use App\Dto\Trait\TimestampableTrait;
 use App\Entity\Tenant\Playlist;
 use App\Entity\Tenant\Screen;
 
 class ScreenCampaign
 {
+    use BlameableTrait;
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply src/Dto/ScreenCampaign.php:13]: Yeah, why not? Fixed in https://github.com/os2display/display-api-service/pull/173/commits/bc2e1b9f06cbd5feb03e96e8614b11dbb1e9b37c.

The traits came about as a desperate attempt to fix something that may be broken in the API (cf. https://github.com/os2display/display-admin-client/pull/223).

**Review by tuj — APPROVED** [review]

**Pull request merged**